validation/virustotal.go

```diff
@@ -31,86 +34,149 @@ type AnalysisResults struct {
 		} `json:"stats,omitempty"`
 		Status string `json:"status"`
 	} `json:"attributes,omitempty"`
+	Meta struct {
+		FileInfo struct {
+			SHA256 string `json:"sha256"`
+		} `json:"file_info"`
+	} `json:"meta"`
+}
+
+type PreviousAnalysisResults struct {
+	Attributes struct {
+		Stats *struct {
+			Suspicious *int `json:"suspicious,omitempty"`
+			Malicious  *int `json:"malicious,omitempty"`
+		} `json:"last_analysis_stats,omitempty"`
+	} `json:"attributes,omitempty"`
 }
 
-func ScanFiles(ctx context.Context, files []io.Reader, names []string) (bool, error) {
+type ScanResult struct {
+	Safe     bool
+	URL      *string
+	Hash     *string
+	FileName string
+}
+
+func ScanFiles(ctx context.Context, files []io.Reader, names []string) ([]ScanResult, error) {
 	errs, gctx := errgroup.WithContext(ctx)
 	fileCount := len(files)
 
-	c := make(chan bool)
+	c := make(chan ScanResult)
 
 	for i := range fileCount {
 		count := i
 		errs.Go(func() error {
-			ok, err := scanFile(gctx, files[count], names[count])
+			scanResult, err := scanFile(gctx, files[count], names[count])
 			if err != nil {
-				return fmt.Errorf("failed to scan file: %w", err)
+				return fmt.Errorf("failed to scan %s: %w", names[count], err)
 			}
-			c <- ok
+			c <- scanResult
 			return nil
 		})
 	}
 	go func() {
 		_ = errs.Wait()
 		close(c)
 	}()
-
-	success := true
+	var results []ScanResult
 	for res := range c {
-		if !res {
-			success = false
-			break
-		}
+		results = append(results, res)
+		// if !res.Safe {
+		// 	break
+		// }
 	}
 
 	if err := errs.Wait(); err != nil {
-		return false, fmt.Errorf("failed to scan file: %w", err)
+		scanResult := []ScanResult{
+			{
+				Safe:     false,
+				FileName: "",
+			},
+		}
+		return scanResult, fmt.Errorf("failed to scan files: %w", err)
 	}
 
-	return success, nil
+	return results, nil
 }
 
-func scanFile(ctx context.Context, file io.Reader, name string) (bool, error) {
-	scan, err := client.NewFileScanner().Scan(file, name, nil)
-	if err != nil {
-		return false, fmt.Errorf("failed to scan file: %w", err)
+func scanFile(ctx context.Context, file io.Reader, name string) (ScanResult, error) {
+	scanResult := ScanResult{
+		Safe:     true,
+		FileName: name,
 	}
 
-	analysisID := scan.ID()
+	hash := sha256.New()
+	if _, err := io.Copy(hash, file); err != nil {
+		return scanResult, fmt.Errorf("failed to generate hash for file %w", err)
+	}
 
-	slox.Info(ctx, "uploaded virus scan", slog.String("file", name), slog.String("analysis_id", analysisID))
+	checksum := hash.Sum(nil)
+	var target PreviousAnalysisResults
```

Comment:
Please name this `analysisResults` or something other than `target` as that is a very special keyword in the codebase.